include/task_manager.h

```diff
@@ -207,6 +214,9 @@ namespace detail {
 		// The last epoch task that has been processed by the executor. Behind a monitor to allow awaiting this change from the main thread.
 		epoch_monitor latest_epoch_reached{initial_epoch_task};
 
+		// The last barrier epoch that was completed from the main thread by calling resume_after_barrier.
+		epoch_monitor latest_barrier_passed{initial_epoch_task};
```

Comment:
Is it necessary to track this independently?

Reply:
This synchronizes the main thread with the executor, but in the opposite direction of `latest_epoch_reached`. The main thread will advance the monitor once it has exited its "capture extraction" region. These events do not occur at the same time, so they can't re-use the monitor.